app/helpers/gsis_controller_helper.rb

```diff
@@ -0,0 +1,156 @@
+    params.require(:gsi).permit(*permitted_parameters)
+  end
+
+  # Creates the employment for the gsi
+  def hire(gsi, hours_per_week)
+    hours_per_week ||= 0
+    @course.gsis << gsi
+    employment(gsi).hours_per_week = hours_per_week || 0
+    employment(gsi).save!
+  end
+
+  # Removes the employment for the gsi
+  def fire(gsi)
+    employment(gsi).destroy!
+  end
+
+  # Whether a new hours/week parameter was passed
+  def hours_per_week_updated
+    hours = params[:gsi][:hours_per_week]
+    hours && (hours != hours_per_week(@gsi))
+  end
+
+  def email_updated
+    new_email = params[:gsi][:email]
+    new_email && (new_email != @gsi.email)
+  end
+
+  # updates how many hours per week the @gsi works
+  def update_employment
+    employment(@gsi).hours_per_week = new_hours_per_week
+    employment(@gsi).save!
+  end
+
+  # updates hours per week for the @gsi or hires him/her
+  def update_hours_per_week
+    if employment(@gsi)
+      update_employment
+    else
+      hire(@gsi, new_hours_per_week)
+    end
+  end
+
+  # sanitizes new hours per week
+  def new_hours_per_week
+    return nil unless params[:gsi][:hours_per_week]
+    params[:gsi][:hours_per_week].to_i || 0
+  end
+
+  # updates email for an existing user or creates a new one
+  # if the @gsi has logged in before or has other appointments
+  def update_email
+    if @gsi.signed_in_before || other_appointments?(@gsi)
+      fire_old_gsi_hire_new_gsi
+    else
+      destroy_old_gsi_hire_new_gsi
+    end
+  end
+
+  # fire the old GSI and hire a new one by email
+  # for the same number of hours/week
+  def fire_old_gsi_hire_new_gsi
+    old_gsi = @gsi
+    new_gsi = find_or_create_by(params[:gsi][:email])
+    if new_gsi.persisted?
+      hours = hours_per_week(old_gsi)
+      hire_and_fire(new_gsi, old_gsi, hours)
+    else
+      render json: @gsi.errors, status: :unprocessable_entity
+    end
+  end
+
+  def hire_and_fire(new_gsi, old_gsi, hours)
+    hire(new_gsi, hours)
+    fire(old_gsi)
+    @gsi = new_gsi
+    @gsi.hours_per_week = hours
+  end
+
+  # destroy the old GSI and hire a new one by email
+  # for the same number of hours/week
+  def destroy_old_gsi_hire_new_gsi
+    old_gsi = @gsi
+    new_gsi = User.find_by(email: params[:gsi][:email])
+    if new_gsi
+      hours = hours_per_week(old_gsi)
+      hire_and_destroy(new_gsi, old_gsi, hours)
+    else
+      update_existing_email
+    end
+  end
+
+  def hire_and_destroy(new_gsi, old_gsi, hours)
+    hire(new_gsi, hours)
+    old_gsi.destroy!
+    @gsi = new_gsi
+    @gsi.hours_per_week = hours
+  end
+
+  # updates the existing GSI's email
+  def update_existing_email
+    unless @gsi.update(model_params)
+      render json: @gsi.errors, status: :unprocessable_entity
+    end
+  end
+
+  # checks if the GSI has appointments other than this one
+  def other_appointments?(gsi)
+    gsi.appointments_count > 1
+  end
+
+  # send an email to the user if their password changed
+  def notify_user(gsi)
```

Comment:
Unused method argument - `gsi`. If it's necessary, use `_` or `_gsi` as an argument name to indicate that it won't be used. You can also write as `notify_user(*)` if you want the method to accept any arguments but don't care about them.